dtls.c

```diff
@@ -1496,10 +1496,14 @@ dtls_prepare_record(dtls_peer_t *peer, dtls_security_parameters_t *security,
     memcpy(A_DATA + 8,  &DTLS_RECORD_HEADER(sendbuf)->content_type, 3); /* type and version */
     dtls_int_to_uint16(A_DATA + 11, res - 8); /* length */
 
-    res = dtls_encrypt(start + 8, res - 8, start + 8, nonce,
-		       dtls_kb_local_write_key(security, peer->role),
-		       dtls_kb_key_size(security, peer->role),
-		       A_DATA, A_DATA_LEN);
+    /* For backwards-compatibility, dtls_encrypt_params is called with
+     * M=<macLen> and L=3. */
+    const dtls_ccm_params_t params = { nonce, 8, 3 };
```

Comment:
To be backwards compatible to really old compilers, we have so far put variable definitions on the beginning of a block. `params` therefore should be moved to the beginning of the function body or enclosed together with the `dtls_encrypt_params()` invocation in `{` ... `}`.
The same holds below in `decrypt_verify()`.

Reply:
OK, I adapt it.

Reply:
The params are move. I removed the const as well, though the intention would be to place the macLength there.

Reply:
if it's OK for you, I squash the commits.

Reply:
I also adapted the "add_ccm" PR. It's possible to get the macLen ahead. so const will also work. I will change it again.

Reply:
Ok, great. Please squash.